src/app/pages/profile-page/profile-page.component.ts

```diff
@@ -40,16 +61,30 @@ export class ProfilePageComponent implements OnInit {
     this.emailMode = 'writing';
   }
 
+  public makeVisibleModal() {
+    this.visibleModal = true;
+  }
+
   public updateUserData(newName: string, newEmail: string): void {
     this.store.dispatch(
       updateUserName({
-        name: newName.length ? newName : this.name,
-        email: newEmail.length ? newEmail : this.email,
+        name: newName.length ? newName : this.currentName,
+        email: newEmail.length ? newEmail : this.currentEmail,
```

Comment:
It seems to me that this approach may lead to the user entering spaces as a name. 
But maybe I'm wrong and trim is applied to fields somewhere else

Reply:
thank you!
I didn't think about it

Reply:
done